### PR TITLE
[10700] Add FASTDDS_STATISTICS CMake option

### PR DESCRIPTION
**Pull request by JLBuenoLopez-eProsima**
New Statistics tool requires a CMake option that enables the building of the specific code implemented for this tool to prevent taxing the performance.

Here the [related documentation PR](https://github.com/eProsima/Fast-DDS-docs/pull/241).

Signed-off-by: JLBuenoLopez-eProsima <joseluisbueno@eprosima.com>

### CODE REVIEW

File: versions.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 Forthcoming
 -----------
 
+* New Fast DDS Statistics module (implies ABI break)
```
Comment (EduPonz) [inline versions.md:4]: I don't think this is necessarily true. Not on the DDS API at least

**Review by EduPonz — CHANGES_REQUESTED** [review]

File: versions.md (hunk 1 above)
Comment (MiguelCompany) [reply versions.md:4]: But we are going to add new methods on the RTPS entities, so the ABI break will be there anyways

Comment (EduPonz) [reply versions.md:4]: I agree, I just won't write it here until they happen. This PR does not entail an ABI break, so I won't state it yet.

Comment (JLBuenoLopez-eProsima) [reply versions.md]: The statement saying that `implies ABI break` has been removed in 19ab1b3d01118aff59fdfc5e958b494b2659a729

**Review by MiguelCompany — APPROVED** [review]

**Pull request merged**